File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 1
    for i in range(2,number+1):
        a=2
        for j in range(2,int(i**0.5)+1):
            if i%j==0:
                if i/j==j: a+=1
                else: a+=2
            if a>limit:
                a=power
                break
        answer+=a
    return answer

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.kt
import kotlin.math.sqrt

class Solution {
    fun numbers(num : Int,limit :Int,power: Int) : Int{
        var count = 0 
        var nums = sqrt(num.toDouble()).toInt()
        for(i in 1 .. nums){
            if(i*i==num) count+=1
            else if(num%i==0) {
                count+=2
            }
            if(count>limit) {
                count = power
                break
            }
            
        }
        return count
    }
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer: Int = 0
        for(i in 0 until number){
            var n = numbers(i+1,limit,power)
            answer+=n
        }
        return answer
    }
}